test/functional/pdo_sqlsrv/pdo_AE_functions.inc

```diff
@@ -0,0 +1,532 @@
+        }
+    }
+
+    $query = substr($query, 0, -6)."\n)";
+
+    return $query;
+}
+
+// The INSERT query for the table
+function formulateSetupQuery($tableName, &$dataTypes, &$colNames, &$colNamesAE)
+{
+    $queryTypes = "(";
+    $valuesString = "VALUES (";
+
+    foreach ($dataTypes as $type) {
+        $colName1 = $colNames[$type].", ";
+        $colName2 = $colNamesAE[$type].", ";
+        $queryTypes .= $colName1;
+        $queryTypes .= $colName2;
+        $valuesString .= "?, ?, ";
+    }
+
+    $queryTypes = substr($queryTypes, 0, -2).")";
+    $valuesString = substr($valuesString, 0, -2).")";
+
+    $insertQuery = "INSERT INTO $tableName ".$queryTypes." ".$valuesString;
+
+    return $insertQuery;
+}
+
+function insertValues($conn, $insertQuery, $dataTypes, $testValues)
+{
+    for ($v = 0; $v < sizeof($testValues['bigint']); ++$v) {
+        $insertValues = array();
+
+        foreach ($dataTypes as $type) {
+            $insertValues[] = $testValues[$type][$v];
+            $insertValues[] = $testValues[$type][$v];
+        }
+
+        // Insert the data using sqlsrv_prepare()
+        try {
+            $stmt = $conn->prepare($insertQuery);
+            $stmt->execute($insertValues);
+        } catch (PDOException $error) {
+            print_r($error);
+            die("Inserting values in encrypted table failed\n");
+        }
+    }
+}
+
+// compareResults checks that the results between the encrypted and non-encrypted
+// columns are identical if statement execution succeeds. If statement execution
+// fails, this function checks for the correct error.
+function compareResults($AEstmt, $nonAEstmt, $key, $encryptionType, $attestation, $comparison='', $type='')
+{
+    try {
+        $nonAEstmt->execute();
+    } catch(Exception $error) {
+        print_r($error);
+        die("Executing non-AE statement failed!\n");
+    }
+
+    try {
+        $AEstmt->execute();
+    } catch(Exception $error) {
+        if ($attestation == 'enabled') {
+            if ($encryptionType == 'Deterministic') {
+                if ($comparison == '=') {
+                    print_r($error);
+                    die("Equality comparison failed for deterministic encryption!\n");
+                } else {
+                    $e = $error->errorInfo;
+                    checkErrors($e, array('42000', '33277'));
+                }
+            } elseif (isEnclaveEnabled($key)) {
+                $e = $error->errorInfo;
+                checkErrors($e, array('42000', '33546'));
+            } elseif (!isEnclaveEnabled($key)) {
+                $e = $error->errorInfo;
+                checkErrors($e, array('42000', '33277'));
+            } else {
+                print_r($error);
+                die("AE statement execution when it shouldn't!");
+            }
```

Comment:
The message for die() is missing a word like "failed"
See another similar one below